core/src/test/scala/org/apache/pekko/persistence/jdbc/state/scaladsl/JdbcDurableStateSpec.scala

```diff
@@ -113,6 +113,45 @@ abstract class JdbcDurableStateSpec(config: Config, schemaType: SchemaType) exte
         }
       }
     }
+    "delete old object revision but not latest" in {
+      whenReady {
+        for {
+
+          n <- stateStoreString.upsertObject("p987", 1, "a valid string", "t123")
+          _ = n shouldBe pekko.Done
+          g <- stateStoreString.getObject("p987")
+          _ = g.value shouldBe Some("a valid string")
+          u <- stateStoreString.upsertObject("p987", 2, "updated valid string", "t123")
+          _ = u shouldBe pekko.Done
+          d <- stateStoreString.deleteObject("p987", 1)
+          _ = d shouldBe pekko.Done
+          h <- stateStoreString.getObject("p987")
+
+        } yield h
+      } { v =>
+        v.value shouldBe Some("updated valid string")
+      }
+    }
+    "delete latest object revision but not older one" in {
+      whenReady {
+        for {
+
+          n <- stateStoreString.upsertObject("p9876", 1, "a valid string", "t123")
+          _ = n shouldBe pekko.Done
+          g <- stateStoreString.getObject("p9876")
+          _ = g.value shouldBe Some("a valid string")
+          u <- stateStoreString.upsertObject("p9876", 2, "updated valid string", "t123")
+          _ = u shouldBe pekko.Done
+          d <- stateStoreString.deleteObject("p9876", 2)
+          _ = d shouldBe pekko.Done
+          h <- stateStoreString.getObject("p9876")
+
+        } yield h
+      } { v =>
+        // TODO current behavior is that deleting the latest revision means getObject returns None (not an older revision)
+        v.value shouldBe None
```

Comment:
I don't think this result is correct but I'm making the result match the observed behavior

Reply:
So far, it seems like we don't actually store the various revisions for the pekko-persistence-jdbc module.

It seems like only the most recently added one is kept. So if you delete an old revision, nothing happens (nothing to delete). If you delete the latest revision, then there is nothing left with that persistenceId.

I will debug this more thoroughly later.

Reply:
@nvollmar You appear to be quite interested in the Cassandra persistence. I'm wondering if you know how the 'revision' is supported in that module.

Reply:
so the 'upsert logic' only updates the existing record if revision > 1

https://github.com/apache/incubator-pekko-persistence-jdbc/blob/7cbcb3f40641a145d1ae5cfbe5ec1e1fef61135a/core/src/main/scala/org/apache/pekko/persistence/jdbc/state/scaladsl/JdbcDurableStateStore.scala#L105

I don't understand why the API has deleteObject(persistenceId, revision) if it does not keep all the revisions.

Reply:
> So far, it seems like we don't actually store the various revisions for the pekko-persistence-jdbc module.

That's why they named it durable state rather than event sourcing, this behavior not a issue.

Reply:
> I don't understand why the API has deleteObject(persistenceId, revision) if it does not keep all the revisions.

I think it is for safety reasons. When a split-brain occurs, nodes in the older version of the partition are not allowed to try to delete the data with the newer version.

For example, we have nodes: [A, B, C, D], and the split-brain splits them into two network partitions of [A,C] [B,D]. 

Node A updated the revision of persistence Actor X001 from 10 to 11, while Node B has Sharding X001 is still version 10, so it is not allowed to delete X001 from B, but it is okay to delete X001 from A.

Reply:
thanks for the explanation @Roiocam